src/fswAlgorithms/attGuidance/prescribedRot2DOF/prescribedRot2DOF.c

```diff
@@ -87,4 +89,156 @@ are updated in this routine as a function of time and written to the prescribedM
 */
 void Update_prescribedRot2DOF(PrescribedRot2DOFConfig *configData, uint64_t callTime, int64_t moduleID)
 {
+    // Create buffer messages
+    HingedRigidBodyMsgPayload spinningBodyRef1In;
+    HingedRigidBodyMsgPayload spinningBodyRef2In;
+    HingedRigidBodyMsgPayload spinningBodyOut;
+    PrescribedMotionMsgPayload prescribedMotionOut;
+
+    // Zero the output messages
+    spinningBodyOut = HingedRigidBodyMsg_C_zeroMsgPayload();
+    prescribedMotionOut = PrescribedMotionMsg_C_zeroMsgPayload();
+
+    // Read the input messages
+    spinningBodyRef1In = HingedRigidBodyMsg_C_zeroMsgPayload();
+    if (HingedRigidBodyMsg_C_isWritten(&configData->spinningBodyRef1InMsg))
+    {
+        spinningBodyRef1In = HingedRigidBodyMsg_C_read(&configData->spinningBodyRef1InMsg);
+    }
+
+    spinningBodyRef2In = HingedRigidBodyMsg_C_zeroMsgPayload();
+    if (HingedRigidBodyMsg_C_isWritten(&configData->spinningBodyRef2InMsg))
+    {
+        spinningBodyRef2In = HingedRigidBodyMsg_C_read(&configData->spinningBodyRef2InMsg);
+    }
+
+    /* This loop is entered when the spinning body attitude converges to the reference attitude. The PRV angle and axis
+     reference parameters are updated along with the profiled trajectory parameters. */
+    if ((HingedRigidBodyMsg_C_timeWritten(&configData->spinningBodyRef1InMsg) <= callTime || HingedRigidBodyMsg_C_timeWritten(&configData->spinningBodyRef2InMsg) <= callTime ) && configData->convergence)
+    {
+        // Define the initial time
+        configData->tInit = callTime * NANO2SEC;
+
+        // Calculate dcm_F0M. This DCM represents the current spinning body attitude with respect to the M frame
+        double dcm_FM[3][3];
+        MRP2C(configData->sigma_FM, dcm_FM);
+        m33Copy(dcm_FM, configData->dcm_F0M);
+
+        // Store the reference variables from the spinningBody input messages
+        double theta1Ref = spinningBodyRef1In.theta;
+        double theta2Ref = spinningBodyRef2In.theta;
+        double thetaDot1Ref = spinningBodyRef1In.thetaDot;
```

Comment:
:small_red_triangle_down: These two variables are never used. Is that intended? Should they be removed?

Reply:
They should be removed. Done.